app/src/main/java/nu/yona/app/ui/message/NotificationFragment.java

```diff
@@ -244,12 +154,160 @@ public void onItemClick(View view)
+	{
+		messageIntentClicked = new Intent(IntentEnum.ACTION_SINGLE_WEEK_DETAIL_VIEW.getActionString());
+		messageIntentClicked.putExtra(AppConstant.YONA_WEEK_DETAIL_URL, yonaMessageClicked.getLinks().getWeekDetails().getHref());
+		setIntentExtrasForActivityCommentMessageClick();
+		YonaAnalytics.createTapEventWithCategory(AnalyticsConstant.NOTIFICATION, AnalyticsConstant.WEEK_ACTIVITY_DETAIL_SCREEN);
+	}
+
+	private void setIntentExtrasForActivityCommentMessageClick()
+	{
+		if (yonaMessageClicked.getLinks().getReplyComment() != null && !TextUtils.isEmpty(yonaMessageClicked.getLinks().getReplyComment().getHref()))
+		{
+			messageIntentClicked.putExtra(AppConstant.YONA_THEME_OBJ, new YonaHeaderTheme(false, null, null, 0, 0, null, R.color.grape, R.drawable.triangle_shadow_grape));
+		}
+		else
+		{
+			messageIntentClicked.putExtra(AppConstant.YONA_BUDDY_OBJ, findBuddy(yonaMessageClicked.getLinks().getYonaBuddy()));
+			messageIntentClicked.putExtra(AppConstant.YONA_THEME_OBJ, new YonaHeaderTheme(true, null, null, 0, 0, null, R.color.mid_blue, R.drawable.triangle_shadow_blue));
+		}
+	}
+
+	private void handleGoalConflictMessageClick()
+	{
+		if (yonaMessageClicked.getLinks().getYonaDayDetails() == null || (TextUtils.isEmpty(yonaMessageClicked.getLinks().getYonaDayDetails().getHref())))
+		{
+			return;
+		}
+		messageIntentClicked = new Intent(IntentEnum.ACTION_SINGLE_ACTIVITY_DETAIL_VIEW.getActionString());
+		messageIntentClicked.putExtra(AppConstant.YONA_DAY_DEATIL_URL, yonaMessageClicked.getLinks().getYonaDayDetails().getHref());
+		messageIntentClicked.putExtra(AppConstant.URL, yonaMessageClicked.getUrl());
+		setEventTimeForGoalConflictMessageIntent();
+		if (yonaMessageClicked.getLinks() != null && yonaMessageClicked.getLinks().getYonaBuddy() != null && !TextUtils.isEmpty(yonaMessageClicked.getLinks().getYonaBuddy().getHref()))
+		{
+			messageIntentClicked.putExtra(AppConstant.YONA_THEME_OBJ, new YonaHeaderTheme(true, null, null, 0, 0, null, R.color.mid_blue_two, R.drawable.triangle_shadow_blue));
+			messageIntentClicked.putExtra(AppConstant.YONA_BUDDY_OBJ, findBuddy(yonaMessageClicked.getLinks().getYonaBuddy()));
+		}
+		else
+		{
+			messageIntentClicked.putExtra(AppConstant.YONA_THEME_OBJ, new YonaHeaderTheme(false, null, null, 0, 0, null, R.color.grape, R.drawable.triangle_shadow_grape));
+		}
+		YonaAnalytics.createTapEventWithCategory(AnalyticsConstant.NOTIFICATION, NotificationEnum.GOALCONFLICTMESSAGE.getNotificationType());
+	}
+
+	private void setEventTimeForGoalConflictMessageIntent()
+	{
+		try
+		{
+			SimpleDateFormat sdf = new SimpleDateFormat(AppConstant.YONA_LONG_DATE_FORMAT, Locale.getDefault());
+			Date date = sdf.parse(yonaMessageClicked.getActivityStartTime());
+			Calendar calendar = Calendar.getInstance();
+			calendar.setTime(date);
+			messageIntentClicked.putExtra(AppConstant.EVENT_TIME, calendar.get(Calendar.HOUR) + ":" + calendar.get(Calendar.MINUTE));
+		}
+		catch (Exception e)
```

Comment:
Don't catch generic ``Exception`` but just catch what's needed (I know it was in the old code already, but please make this improvement along with the other improvements you're making).